contrib/storage-http/src/main/java/org/apache/drill/exec/store/http/HttpStoragePluginConfig.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill.exec.store.http;
+
+import org.apache.drill.common.map.CaseInsensitiveMap;
+import org.apache.drill.common.logical.StoragePluginConfigBase;
+import org.apache.drill.shaded.guava.com.google.common.base.Objects;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+
+@JsonTypeName(HttpStoragePluginConfig.NAME)
+public class HttpStoragePluginConfig extends StoragePluginConfigBase {
+  private static final Logger logger = LoggerFactory.getLogger(HttpStoragePluginConfig.class);
+
+  public static final String NAME = "http";
+
+  public final Map<String, HttpAPIConfig> connections;
```

Comment:
The idea here is to have one HTTP plugin config with multiple connections. I wonder if this is useful in practice? When would I create a new connection and when a new plugin config?

Might it make sense for each plugin config to correspond to one HTTP service? Flatten the `HttpAPIConfig` properties here.

Then, in my query, rather than saying:

```
SELECT * FROM http.google.foo;
SELECT * FROM http.facebook.bar;
```

I might say:

```
SELECT * FROM googleRest.foo;
SELECT * FROM facebookRest.bar;
```

Reply:
Also, as designed, there is no list of endpoints. That is, how do I specify that, in the example above, `foo` is a valid endpoint for Google (`http:/something.google.com/foo`) while `bar` is a valid endpoint for Facebook?

One would expect there to be multiple endpoints: `users`, `posts`, `sales`, `products`, etc. At present, these are not declared in the config. Must the user know which ones are valid?

Reply:
Many REST calls accept parameters. I might have a `users` endpoint that returns all users, and I might have an a `userOrders` endpoint that needs a user parameter: `userOrders?user=bob`. Do we support this concept in this version of the plugin?

Since we don't specify endpoints (REST services), we don't seem to have a place to specify parameters.

Reply:
@paul-rogers 
I sent a note to the email alias about this.  Here is how I envisioned this being used: a user could create two instances of this plugin, one called `FaceSpace` and the other called `CatTube`.  Within in each, they could define the various APIs within that space.  So for instance, they could have:
```sql
SELECT *
FROM `cattube`.`kittens`.`?param1=foo&param2=bar`
```
and
```sql
SELECT * 
FROM `CatTube`.`cuteCats`.`?param1=foo&param2=bar`
```
Alternatively, they could group them all into one instance, and give the connections names like `CatTube_cuteKittens` etc.

The parameters make up the `table` portion of the query.  Of course static parameters can be passed in the config as well as in post bodies etc, all in the config. 

Does that make sense?